src/UIShell/HeaderGlobalAction.svelte

```diff
@@ -10,17 +14,18 @@
 
   /** Obtain a reference to the HTML button element */
   export let ref = null;
+
+  import Button from "../Button/Button.svelte";
+
+  $: buttonClass = [
+    "bx--header__action",
+    isActive && " bx--header__action--active",
+    $$restProps.class,
+  ]
+    .filter(Boolean)
+    .join(" ");
 </script>
 
-<button
-  type="button"
-  bind:this="{ref}"
-  class:bx--header__action="{true}"
-  class:bx--header__action--active="{isActive}"
-  {...$$restProps}
-  on:click
->
-  <slot>
-    <svelte:component this="{icon}" size="{20}" />
-  </slot>
-</button>
+<Button bind:this="{ref}" class="{buttonClass}" {...$$restProps} on:click>
```

Comment:
The order of the properties matters and probably causes issues.

`class:` directives work independently from a `class` property passed in via `$$restProps`.
But now that a component is used, I suspect that when a `class` is set, the spread will overwrite what was set before to `buttonClass`.

If that is the case, the spread should happen before setting class or the component should define a `class` property, so it no longer is part of `$$restProps`.

Reply:
I did not find any differences depending on the order of these props, but just in case I put $$restProps before class.